src/CognitoDatasetManager.js

```diff
@@ -304,56 +195,378 @@ class CognitoDatasetManager extends EventEmitter {
+            sampleName[2].match("(.*)\\([0-9]*\\)$")[1] +
+            "(" +
+            v.toString() +
+            ")" +
+            "." +
+            sampleName[3]
+        } else {
+          sampleName[1] =
+            sampleName[2] + "(" + v.toString() + ")." + sampleName[3]
+        }
+        v++
+      } else {
+        boolName = false
+      }
+    }
+
+    return sampleName[1]
+  }
+
+  getSampleWithName = (samples, sampleName) => {
+    samples = samples.filter((sample) => {
+      if (this.getSampleName(sample) === sampleName) return true
+      return false
+    })
+    return samples[0]
+  }
+
+  getSampleUrl = (sample) => {
+    if (!sample) return
+    return (
+      sample.imageUrl ||
+      sample.videoUrl ||
+      sample.audioUrl ||
+      sample.pdfUrl ||
+      sample.txtUrl ||
+      sample.timeUrl ||
+      undefined
+    )
+  }
+
+  setSampleUrl = async (sample, newUrl) => {
+    if (!sample) return
+    if (sample.imageUrl) sample = await setIn(sample, ["imageUrl"], newUrl)
+    if (sample.videoUrl) sample = await setIn(sample, ["videoUrl"], newUrl)
+    if (sample.audioUrl) sample = await setIn(sample, ["audioUrl"], newUrl)
+    if (sample.pdfUrl) sample = await setIn(sample, ["pdfUrl"], newUrl)
+    if (sample.txtUrl) sample = await setIn(sample, ["txtUrl"], newUrl)
+    if (sample.timeUrl) sample = await setIn(sample, ["timeUrl"], newUrl)
+    return sample
+  }
+
+  getAssetText = async (sample) => {
+    if (sample.document) {
+      return sample.document
+    }
+
+    if (sample.txtUrl) {
+      var response = await fetch(sample.txtUrl, {
+        method: "GET",
+      }).catch((error) => {
+        console.log("Looks like there was a problem: \n", error)
+      })
+      const texte = await response.text()
+      return texte
+    }
+
+    return undefined
+  }
+  getSampleText = this.getAssetText
+
+  getAssetTime = async (sample) => {
+    if (sample.timeData) {
+      return { timeData: sample.timeData }
+    }
+
+    if (sample.timeUrl) {
+      var response = await fetch(sample.timeUrl, {
+        method: "GET",
+      }).catch((error) => {
+        console.log("Looks like there was a problem: \n", error)
+      })
+      const json = await response.json()
+      return json
+    }
+
+    return undefined
+  }
+  getSampleTime = this.getAssetTime
+
+  getSampleNameFromURL = (sample) => {
+    var sampleName
+    if (!isEmpty(sample)) {
+      if (!isEmpty(this.getSampleUrl(sample))) {
+        sampleName = decodeURI(this.getSampleUrl(sample)).match(
+          `.*\\/(([^\\/\\\\&\\?]*)\\.([a-zA-Z0-9]*))(\\?|$)`
+        )
+      }
+    }
+    return sampleName
+  }
+
+  getSampleExtensionFromURL = (sample) => {
+    var sampleName
+    if (!isEmpty(sample)) {
+      if (!isEmpty(this.getSampleUrl(sample))) {
+        sampleName = decodeURI(this.getSampleUrl(sample)).match(
+          `[^\\/\\\\&\\?]*\\.[a-zA-Z0-9]*(?:\\?|$)`
+        )
+      }
+    }
+    return sampleName
+  }
```

Comment:
we'd want to move these into a base class or utility function

Reply:
If you talk about :
getSampleExtensionFromURL,
getSampleNameFromURL,
setSampleUrl
getSampleUrl
getSampleWithName

It's the idea